cw_tron/lib/tron_wallet.dart

```diff
@@ -0,0 +1,557 @@
+
+      if (shouldSendAll) {
+        totalAmount = walletBalanceForCurrency;
+      } else {
+        final totalOriginalAmount = double.parse(output.cryptoAmount ?? '0.0');
+        totalAmount =TronHelper.toSun(totalOriginalAmount.toString());
+      }
+
+      if (walletBalanceForCurrency < totalAmount || totalAmount < BigInt.zero) {
+        throw TronTransactionCreationException(transactionCurrency);
+      }
+    }
+
+    final tronBalance = balance[CryptoCurrency.trx]?.balance ?? BigInt.zero;
+
+    final pendingTransaction = await _client.signTransaction(
+      ownerPrivKey: _tronPrivateKey,
+      toAddress: tronCredentials.outputs.first.isParsedAddress
+          ? tronCredentials.outputs.first.extractedAddress!
+          : tronCredentials.outputs.first.address,
+      amount: TronHelper.fromSun(totalAmount),
+      currency: transactionCurrency,
+      tronBalance: tronBalance,
+      sendAll: shouldSendAll,
+    );
+
+    return pendingTransaction;
+  }
+
+  @override
+  Future<Map<String, TronTransactionInfo>> fetchTransactions() async => {};
+
+  Future<void> fetchAllTransactions() async {
+    final address = _tronAddress;
+
+    final transactions = await _client.fetchTransactions(address);
+
+    final Map<String, TronTransactionInfo> result = {};
+
+    final contract = ContractABI.fromJson(trc20Abi, isTron: true);
+
+    final ownerAddress = TronAddress(_tronAddress);
+
+    for (var transactionModel in transactions) {
+      if (transactionModel.isError) {
+        continue;
+      }
+
+      String? tokenSymbol;
+      if (transactionModel.contractAddress != null) {
+        final tokenAddress = TronAddress(transactionModel.contractAddress!);
+
+        tokenSymbol = (await _client.getTokenDetail(
+              contract,
+              "symbol",
+              ownerAddress,
+              tokenAddress,
+            ) as String?) ??
+            '';
+      }
+
+      result[transactionModel.hash] = TronTransactionInfo(
+        id: transactionModel.hash,
+        tronAmount: transactionModel.amount ?? BigInt.zero,
+        direction: TronAddress(transactionModel.from!, visible: false).toAddress() == address
+            ? TransactionDirection.outgoing
+            : TransactionDirection.incoming,
+        blockTime: transactionModel.date,
+        txFee: transactionModel.fee,
+        tokenSymbol: tokenSymbol ?? "TRX",
+        to: transactionModel.to,
+        from: transactionModel.from,
+        isPending: false,
+      );
+    }
+
+    transactionHistory.addMany(result);
+
+    await transactionHistory.save();
+  }
+
+  Future<void> fetchTrc20ExcludedTransactions() async {
+    final address = _tronAddress;
+
+    final transactions = await _client.fetchTrc20ExcludedTransactions(address);
+
+    final Map<String, TronTransactionInfo> result = {};
+
+    for (var transactionModel in transactions) {
+      if (transactionHistory.transactions.containsKey(transactionModel.hash)) {
+        continue;
```

Comment:
this should fill in any missing data from the already existing transaction

Reply:
So I went over it over our call and its actually the right way to handle it

So we have the transactionsHistory already fetched with the initial data, this particular call, comes to fill in the missing data, but we don't want to duplicate, when we see that the transaction already exist in the history, we continue to the next iteration, we don't re-add the data to the history again as that would cause a duplicate entry.